performance-test/src/gatling/java/org/opensearch/dataprepper/test/performance/tools/Templates.java

```diff
@@ -30,19 +32,31 @@ public static String now() {
     }
     
     public static Function<Session, String> apacheCommonLogTemplate(final int batchSize) {
+        return generateLogArray(batchSize,
+                () -> ipAddress() + " - frank [" + now() + "] \\\"" + httpMethod() + " /apache_pb.gif HTTP/1.0\\\" "+ statusCode() + " " + responseSize());
+    }
+
+    public static Function<Session, String> userAgent(final int batchSize) {
+        return generateLogArray(batchSize, () -> userAgent());
+    }
+
+    private static Function<Session, String> generateLogArray(final int batchSize, final Supplier<String> stringSupplier) {
         return session -> {
             final List<String> logs = IntStream.range(0, batchSize)
-                    .mapToObj(i -> "{\"log\": \"" + ipAddress() + " - frank [" + now() + "] \\\"" + httpMethod() + " /apache_pb.gif HTTP/1.0\\\" "+ statusCode() + " " + responseSize() + "\"}")
+                    .mapToObj(i -> "{\"log\": \"" + stringSupplier.get() + "\"}")
                     .collect(Collectors.toList());
-            final String logArray = String.join(",", logs);
-            return "[" + logArray + "]";
+            return logs.stream().collect(Collectors.joining(",", "[", "]"));
```

Comment:
Why do you materialize the stream to a collection to stream it again? Can't you just do the last `collect`?

Reply:
Good catch, thanks! This was just a miss from my refactoring the code. I just pushed a change to simplify this.